sdk/provisioning/Generator/src/Specifications/StorageSpecification.cs

```diff
@@ -0,0 +1,63 @@
+        Roles.Add(new Role("StorageFileDataSmbShareContributor", "0c867c2a-1d8c-454a-a3db-ab2ea1bdc8bb", "Allows for read, write, and delete access on files/directories in Azure file shares. This role has no built-in equivalent on Windows file servers."));
+        Roles.Add(new Role("StorageFileDataSmbShareElevatedContributor", "a7264617-510b-434b-a828-9731dc254ea7", "Allows for read, write, delete, and modify ACLs on files/directories in Azure file shares. This role is equivalent to a file share ACL of change on Windows file servers."));
+        Roles.Add(new Role("StorageFileDataSmbShareReader", "aba4ae5f-2193-4029-9191-0cb91df5e314", "Allows for read access on files/directories in Azure file shares. This role is equivalent to a file share ACL of read on Windows file servers."));
+        Roles.Add(new Role("StorageQueueDataContributor", "974c5e8b-45b9-4653-ba55-5f855dd0fb88", "Read, write, and delete Azure Storage queues and queue messages. To learn which actions are required for a given data operation, see Permissions for calling data operations."));
+        Roles.Add(new Role("StorageQueueDataMessageProcessor", "8a0f0c08-91a1-4084-bc3d-661d67233fed", "Peek, retrieve, and delete a message from an Azure Storage queue. To learn which actions are required for a given data operation, see Permissions for calling data operations."));
+        Roles.Add(new Role("StorageQueueDataMessageSender", "c6a89b2d-59bc-44d0-9896-0f6e12d7b80a", "Add messages to an Azure Storage queue. To learn which actions are required for a given data operation, see Permissions for calling data operations."));
+        Roles.Add(new Role("StorageQueueDataReader", "19e7f393-937e-4f77-808e-94535e297925", "Read and list Azure Storage queues and queue messages. To learn which actions are required for a given data operation, see Permissions for calling data operations."));
+        Roles.Add(new Role("StorageTableDataContributor", "0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3", "Allows for read, write and delete access to Azure Storage tables and entities"));
+        Roles.Add(new Role("StorageTableDataReader", "76199698-9eea-4c19-bc75-cec21354c6b6", "Allows for read access to Azure Storage tables and entities"));
+        Roles.Add(new Role("ClassicStorageAccountContributor", "86e8f5dc-a6e9-4c67-9d15-de283e8eac25", "Lets you manage classic storage accounts, but not access to them."));
+        Roles.Add(new Role("ClassicStorageAccountKeyOperatorServiceRole", "985d6b00-f706-48f5-a6fe-d0ca12fb668d", "Classic Storage Account Key Operators are allowed to list and regenerate keys on Classic Storage Accounts"));
+
+        // Assign Roles
+        CustomizeResource<StorageAccountResource>(r => r.GenerateRoleAssignment = true);
```

Comment:
The more interesting part is what it takes to spec a new service.  There's a little bit of tedium, but it's very quick.  We just generate everything, tweak stuff by hand if the generator gets it wrong, and add in extra metadata like the `Roles` or which resources we want an `AssignRole` extension on.  The eventual goal is to transition all of this into TypeSpec decorators we can use to augment the management plane APIs and generate across multiple languages.  I'm going to push back on anything that makes this logic overly complicated.